draft-ietf-quic-transport.md

```diff
@@ -1451,33 +1454,279 @@ failure.  If integrity protection is performed by QUIC, QUIC MUST abort the
+rate. Therefore, the endpoint resets its congestion controller, as described in
+{{migration-cc}}.
+
+Receiving acknowledgments for this data serves as proof of the peer's
+reachability from the new address.  Note that since acknowledgments may be
+received on any path, return reachability on the new path is not
+established. To establish return reachability on the new path, an endpoint MAY
+concurrently initiate path validation {{migrate-validate}} on the new path.
+
+
+### Responding to Connection Migration {#migration-response}
+
+Receiving a packet from a new peer address containing a non-probing frame
+indicates that the peer has migrated to that address.
+
+In response to such a packet, an endpoint MUST start sending subsequent packets
+to the new peer address and MUST initiate path validation ({{migrate-validate}})
+to verify the peer's ownership of the unvalidated address.
+
+An endpoint MAY send data to an unvalidated peer address, but it MUST protect
+against potential attacks as described in {{address-spoofing}} and
+{{on-path-spoofing}}.  An endpoint MAY skip validation of a peer address if that
+address has been seen recently.
+
+An endpoint only changes the address that it sends packets to in response to the
+highest-numbered non-probing packet. This ensures that an endpoint does not send
+packets to an old peer address in the case that it receives reordered packets.
+
+After changing the address to which it sends non-probing packets, an endpoint
+could abandon any path validation for other addresses.
+
+Receiving a packet from a new peer address might be the result of a NAT
+rebinding at the peer.
+
+After verifying a new client address, the server SHOULD send new address
+validation tokens ({{address-validation}}) to the client.
+
+
+#### Handling Address Spoofing by a Peer {#address-spoofing}
+
+It is possible that a peer is spoofing its source address to cause an endpoint
+to send excessive amounts of data to an unwilling host.  If the endpoint sends
+significantly more data than the spoofing peer, connection migration might be
+used to amplify the volume of data that an attacker can generate toward a
+victim.
+
+As described in {{migration-response}}, an endpoint is required to validate a
+peer's new address to confirm the peer's possession of the new address.  Until a
+peer's address is deemed valid, an endpoint MUST limit the rate at which it
+sends data to this address.  The endpoint MUST NOT send more than a minimum
+congestion window's worth of data per estimated round-trip time (as defined in
+{{QUIC-RECOVERY}}).  In the absence of this limit, an endpoint risks being used
+for a denial of service attack against an unsuspecting victim.  Note that since
+the endpoint will not have any round-trip time measurements to this address, the
+estimate SHOULD be the default initial value (see {{QUIC-RECOVERY}}).
+
+An endpoint SHOULD NOT apply this rate limit when the endpoint skips validation
```

Comment:
Remove the "SHOULD NOT" (no interop requirement here).  Maybe:

If an endpoint skips validation of a peer address as described in {{migration-responses}}, it does not need to limit its sending rate.